pint_xarray/accessors.py

```diff
@@ -154,77 +137,97 @@ def quantify(self, units=None, unit_registry=None, registry_kwargs=None):
         `.attrs`. Will raise a ValueError if the DataArray already contains a
         unit-aware array.
 
+        .. note::
+            Be aware that unless you're using ``dask`` this will load
+            the data into memory. To avoid that, consider converting
+            to ``dask`` first (e.g. using ``chunk``).
+
         Parameters
         ----------
-        units : pint.Unit or str, optional
-            Physical units to use for this DataArray. If not provided, will try
-            to read them from `DataArray.attrs['units']` using pint's parser.
-        unit_registry : `pint.UnitRegistry`, optional
+        units : pint.Unit or str or mapping of hashable to pint.Unit or str, optional
+            Physical units to use for this DataArray. If a str or
+            pint.Unit, will be used as the DataArray's units. If a
+            dict-like, it should map a variable name to the desired
+            unit (use the DataArray's name to refer to its data). If
+            not provided, will try to read them from
+            ``DataArray.attrs['units']`` using pint's parser.
```

Comment:
Perhaps good to comment something to the effect of "this will remove the 'units' attribute from DataArray.attrs"?